src/main/java/org/broadinstitute/hellbender/tools/exome/SubtractCoverageComponents.java

```diff
@@ -0,0 +1,277 @@
+
+    @Argument(
+            doc = "PCA input file",
+            fullName = PCA_INPUT_FULL_NAME,
+            shortName = PCA_INPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File pcaFile;
+
+    @Argument(
+            doc = "Normalized target coverage output",
+            fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            optional = false
+    )
+    protected File outputFile;
+
+
+    @Override
+    protected Object doWork() {
+        final PCA pca = readPcaInputFile(pcaFile);
+        final ReadCountCollection coverage = readInputCoverage(inputFile);
+
+        // Match up the targets in the PCA and the input counts file.
+        final Map<String, Integer> pcaTargetIndexByName = composeComponentTargetIndexMap(pca);
+        final Map<String, Integer> coverageTargetIndexByName = composeCoverageTargetIndexMap(coverage);
+        compareCompomentsAndCoverageTargetNames(pcaTargetIndexByName, coverageTargetIndexByName);
+
+
+        // The actual work:
+
+        // T x C matrix.
+        final RealMatrix eigenVectors = pca.getEigenVectors();
+        final int componentCount = eigenVectors.getColumnDimension();
+        final RealVector centers = pca.getCenters();
+        // S x T matrix with the counts ready to be multiplied by the eigen-vectors T x C
+        final RealMatrix transposedCoverage = composeCoverageMatrixReadyForNormalization(pcaTargetIndexByName, coverageTargetIndexByName, centers, coverage);
+        // S x C matrix where each row indicate the projection value of the sample onto a component.
+        final RealMatrix projection = transposedCoverage.multiply(eigenVectors);
+
+        // T x S matrix with the normalized new counts.
+        final RealMatrix resultMatrix = subtractComponentProjectionsAndTranspose(projection, transposedCoverage, eigenVectors);
+
+        // Creates the output read count collection.
+        final ReadCountCollection normalizedCoverage = new ReadCountCollection(
+                SetUniqueList.setUniqueList(composeTargetList(pca, coverage, coverageTargetIndexByName)),
+                SetUniqueList.setUniqueList(new ArrayList<>(coverage.columnNames())), resultMatrix);
+
+        // Output the result normalized coverage.
+        writeOutputCoverage(normalizedCoverage);
+
+        return "SUCCESS";
+    }
+
+    private ReadCountCollection readInputCoverage(final File inputFile) {
+        final ReadCountCollection coverage;
+        try {
+            coverage = ReadCountCollectionUtils.parse(inputFile);
+        } catch (final IOException ex) {
+            throw new UserException.CouldNotReadInputFile(inputFile, ex);
+        }
+        return coverage;
+    }
+
+    private PCA readPcaInputFile(final File pcaFile) {
+        final PCA pca;
+        try (final HDF5File pcaHd5 = new HDF5File(pcaFile, HDF5File.OpenMode.READ_ONLY)) {
+            pca = PCA.readHDF5(pcaHd5);
+        }
```

Comment:
It will call close on pcaHd5 when exiting the try even if readHDF5 resulted in an exception.